src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala

```diff
@@ -50,14 +52,13 @@ class BitbucketClient(key: String, secretKey: String, token: String, secretToken
   }
 
   /*
-   * Does an API post
+   * Does an API request
    */
-  private def post[D,T](request: Request[T], values: D)(implicit reader: Reads[T], writer: Writeable[D], contentType: ContentTypeOf[D]): RequestResponse[T] = withClientRequest { client =>
-
+  private def performRequest[D, T](method: String, request: Request[T], values: D)(implicit reader: Reads[T], writer: Writeable[D], contentType: ContentTypeOf[D]): RequestResponse[T] = withClientRequest { client =>
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [performRequest should have a smaller scope by using private[this]](https://www.codacy.com/app/Codacy/bitbucket-scala-client/file/1807008534/issues/source?bid=2474672&fileBranchId=3073963#l57)